keycloak/openid_client.go

```diff
@@ -80,3 +106,69 @@ func (keycloakClient *KeycloakClient) UpdateOpenidClient(client *OpenidClient) e
 func (keycloakClient *KeycloakClient) DeleteOpenidClient(realmId, id string) error {
 	return keycloakClient.delete(fmt.Sprintf("/realms/%s/clients/%s", realmId, id))
 }
+
+func (keycloakClient *KeycloakClient) GetOpenidClientDefaultScopes(realmId, clientId string) ([]*OpenidClientScope, error) {
+	var scopes []*OpenidClientScope
+
+	err := keycloakClient.get(fmt.Sprintf("/realms/%s/clients/%s/default-client-scopes", realmId, clientId), &scopes)
+	if err != nil {
+		return nil, err
+	}
+
+	return scopes, nil
+}
+
+func (keycloakClient *KeycloakClient) AttachOpenidClientDefaultScopes(realmId, clientId string, scopeNames []interface{}) error {
+	openidClient, err := keycloakClient.GetOpenidClient(realmId, clientId)
+	if err != nil && ErrorIs404(err) {
+		return fmt.Errorf("validation error: client with id %s does not exist", clientId)
+	} else if err != nil {
+		return err
+	}
+
+	if openidClient.BearerOnly {
+		return fmt.Errorf("validation error: client with id %s uses access type BEARER-ONLY which does not use scopes", clientId)
+	}
+
+	allOpenidClientScopes, err := keycloakClient.listOpenidClientScopes(realmId)
+	if err != nil {
+		return err
+	}
+
+	for _, scopeName := range scopeNames {
```

Comment:
this block looks very similar to the one that starts on L160. Is there a way to extract some bits of this to combine them?